user/hubers_functions.hpp

```diff
@@ -687,7 +883,118 @@ namespace antok {
+						         :_eAddr(eAddr), _xAddr(xAddr), _yAddr(yAddr), _zAddr(zAddr),
+						          _xPVAddr(xPVAddr), _yPVAddr(yPVAddr), _zPVAddr(zPVAddr),
+						          _selectedMass(selectedMass), _outLVAddr1(outLVAddr1), _outMAddr1(outMAddr1) , _outLVAddr2(outLVAddr2), _outMAddr2(outMAddr2){}
+
+						virtual ~GetClosestPi0Pi0() {}
+
+						bool operator() () {
+							_outLVAddr1->SetPxPyPzE(9999, 9999, 9999, 99999);
+							*_outMAddr1 = -99;
+							_outLVAddr2->SetPxPyPzE(9999, 9999, 9999, 99999);
+							*_outMAddr2 = -99;
+							double minDist=9e9;
+							int idx2 = 0;
+							if(_eAddr->size() < 4 )
+								return true;
+							{
+								TVector3 v3a( (*_xAddr)[0] - *_xPVAddr, (*_yAddr)[0] - *_yPVAddr, (*_zAddr)[0] - *_zPVAddr) ;
+								v3a.SetMag((*_eAddr)[0]);
+								TLorentzVector lva(v3a, (*_eAddr)[0]);
+								for(unsigned int j = 1; j < _eAddr->size(); ++j) {
+									if( (*_eAddr)[j] < 2 )
+										continue;
+									TVector3 v3b( (*_xAddr)[j] - *_xPVAddr, (*_yAddr)[j] - *_yPVAddr, (*_zAddr)[j] - *_zPVAddr) ;
+									v3b.SetMag((*_eAddr)[j]);
+									TLorentzVector lvb(v3b, (*_eAddr)[j]);
+									if(std::fabs((lva + lvb).Mag() - *_selectedMass < minDist)) {
+										minDist = std::fabs((lva + lvb).Mag() - *_selectedMass);
+										*_outLVAddr1 = lva + lvb;
+										*_outMAddr1 = _outLVAddr1->Mag();
+										idx2 = j;
+									}
+								}
+							}
+							{
+								minDist=9e9;
+								for(unsigned int j = 1; j < _eAddr->size(); ++j) {
+									if(j == idx2)
+										continue;
+									if( (*_eAddr)[j] < 2 )
+										continue;
+									TVector3 v3a( (*_xAddr)[j] - *_xPVAddr, (*_yAddr)[j] - *_yPVAddr, (*_zAddr)[j] - *_zPVAddr) ;
+									v3a.SetMag((*_eAddr)[j]);
+									TLorentzVector lva(v3a, (*_eAddr)[j]);
+									for(unsigned int jj = j + 1; jj < _eAddr->size(); ++jj){
+										if(jj == idx2)
+											continue;
+										if( (*_eAddr)[jj] < 2 )
+											continue;
+										TVector3 v3b( (*_xAddr)[jj] - *_xPVAddr, (*_yAddr)[jj] - *_yPVAddr, (*_zAddr)[jj] - *_zPVAddr) ;
+										v3b.SetMag((*_eAddr)[jj]);
+										TLorentzVector lvb(v3b, (*_eAddr)[jj]);
+										if(std::fabs((lva + lvb).Mag() - *_selectedMass < minDist)) {
+											minDist = std::fabs((lva + lvb).Mag() - *_selectedMass);
+											*_outLVAddr2 = lva + lvb;
+											*_outMAddr2 = _outLVAddr2->Mag();
+										}
+									}
+								}
+							}
+							return true;
+						}
+
+					private:
+						std::vector<double>* _eAddr;
+						std::vector<double>* _xAddr;
+						std::vector<double>* _yAddr;
+						std::vector<double>* _zAddr;
+						double* _xPVAddr;
+						double* _yPVAddr;
+						double* _zPVAddr;
+						double* _selectedMass;
+						TLorentzVector* _outLVAddr1;
+						double* _outMAddr1;
+						TLorentzVector* _outLVAddr2;
+						double* _outMAddr2;
+				};
+
+				//***********************************
+				//Gets run and spill as one double
```

Comment:
why as a double? all numbers are integers, and the numbers are small enough even for signed integers, let alone long longs